File: LC896.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {

    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    int l = nums.size();

    if(l < 3)
        return true;
    
    bool in = false, de = false;

    for(int i = 0; i < l-1; i++){
        if(nums[i] > nums[i+1])
            de = true;
        if(nums[i] < nums[i+1])
            in = true;
        
        if(in && de)
            return false;
    }

    return true;
    }
};

File: LC896.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        n = len(nums)

        if n < 3:
            return True
        
        inc = False
        de = False
        
        for i in range(n-1):
            if nums[i] > nums[i+1]:
                de = True
            if nums[i] < nums[i+1]:
                inc = True
            
            if inc and de:
                return False
        
        return True
